compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1459,11 +1465,17 @@ fn print_native_static_libs(
             lib_args.push(format!("-l{}", lib));
         }
     }
-    if !lib_args.is_empty() {
-        sess.emit_note(errors::StaticLibraryNativeArtifacts);
-        // Prefix for greppability
-        // Note: This must not be translated as tools are allowed to depend on this exact string.
-        sess.note_without_error(format!("native-static-libs: {}", &lib_args.join(" ")));
+
+    match out {
+        OutFileName::Real(_) => out.overwrite(&lib_args.join(" "), sess),
```

Comment:
I feel that just writing the arguments here could potentially be confusing without the output that `emit_note(errors::StaticLibraryNativeArtifacts)` would print to stdout. Can we either include that note or maybe don't write anything here at all?

Reply:
Good call. How about keeping a `sess.emit_note` that informs you that it wrote the native-static-libs to a file? Similar to what #76843 did for long type names -- write the unadorned information to a file (which is the most useful for tooling) and also write an informative note among the compiler diagnostics to provide context.

- **`rustc --print=native-static-libs ...`** (same as today)

    ```console
    note: Link against the following native artifacts when linking against this static library. The order and any duplication can be significant on some platforms.

    note: native-static-libs: -lglib-2.0 -lbar_cli -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc -lsystemd -lfoo_cli
   ```

- **`rustc --print=native-static-libs=./native-static-libs.txt ...`**

    ```console
    note: Native artifacts to link against have been written to ./native-static-libs.txt. The order and any duplication can be significant on some platforms.
    ```

    native-static-libs.txt:

    ```txt
    -lglib-2.0 -lbar_cli -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc -lsystemd -lfoo_cli
    ```